### PR TITLE
(maint) Fix linking errors when building with Leatherman locale

### PR DESCRIPTION
**Pull request by Magisus**
When using leatherman::locale, it is necessary to also link against
boost::locale. On OSX, this further requires linking against the iconv
library.

@MikaelSmith would you mind looking over this and seeing if any of the added bits jump out at you as being definitely unnecessary. This went through enough iterations that I think some things may have been added that are actually irrelevant (and if you find any, I'd love to hear why they're not needed. Still trying to get the full picture on this stuff).

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Yeah, I think this is unfortunately required given how the leatherman/locale/locale.hpp header is structured.

File: lib/tests/CMakeLists.txt [hunk 1]
```diff
@@ -18,16 +18,22 @@ set(TEST_CASES
...
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND BOOST_STATIC)
+    target_link_libraries(lib${PROJECT_NAME}_test iconv)
+endif()
+
 target_link_libraries(lib${PROJECT_NAME}_test
     ${Boost_LIBRARIES}
-    ${LEATHERMAN_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
+    ${LEATHERMAN_LIBRARIES}
```
Comment (MikaelSmith) [inline lib/tests/CMakeLists.txt]: Why was this change necessary?

Comment (MikaelSmith) [inline lib/tests/CMakeLists.txt]: Link order matters. I'd put iconv after Boost_LIBRARIES (probably doesn't matter where it is related to LEATHERMAN_LIBRARIES, but after them as well makes sense to me).

File: lib/tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Setup compiling the test executable. C++ compile flags are inherited from the parent directory.
-include_directories(${LEATHERMAN_CATCH_INCLUDE})
+include_directories(../inc ${LEATHERMAN_CATCH_INCLUDE} ${Boost_INCLUDE_DIRS})
```
Comment (MikaelSmith) [inline lib/tests/CMakeLists.txt]: How did this work before?

File: lib/tests/CMakeLists.txt [hunk 3]
```diff
@@ -18,16 +18,22 @@ set(TEST_CASES
     program_options.cc
 )
 
+link_directories(${Boost_LIBRARY_DIRS})
```
Comment (MikaelSmith) [inline lib/tests/CMakeLists.txt]: This should be implied by find_package(Boost).

File: lib/CMakeLists.txt [hunk 4]
```diff
@@ -70,6 +70,7 @@ set(PROJECT_SOURCES
     src/values/config_concatenation.cc
     src/config_value_factory.cc)
 
+link_directories(${Boost_LIBRARY_DIRS})
```
Comment (MikaelSmith) [inline lib/CMakeLists.txt]: Should also be unnecessary.

File: lib/tests/CMakeLists.txt (hunk 2 above)
Comment (MikaelSmith) [inline lib/tests/CMakeLists.txt]: Oh, the changes here aren't needed, they're inherited from the parent (lib/CMakeLists.txt).

**Comment by Magisus** [comment]
Looks like it still works with the changes you suggested.
